strawberry/types/type_extension.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Sequence, Type
+
+from strawberry.types.types import StrawberryObjectDefinition
+
+if TYPE_CHECKING:
+    from dataclasses import Field
+
+    from graphql import GraphQLAbstractType, GraphQLResolveInfo
+
+    from strawberry.field import StrawberryField
+    from strawberry.type import WithStrawberryObjectDefinition
+
+
+class TypeExtension:
+    def before_wrap_dataclass(self, cls: Type) -> None:
+        pass
```

Comment:
Maybe we should use something similar to our extensions api and define this as `on_wrap_dataclass`, in which the user can `yield` inside, meaning he can call "before" code before it, and "after" code after it. That also allows they to except exceptions

What do you think?